packages/fxa-admin-panel/server/lib/server.ts

```diff
@@ -2,30 +2,58 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// TO DO: CORS, other security-related tasks (#4312)
+
 import express from 'express';
+import bodyParser from 'body-parser';
 import path from 'path';
 import serveStatic from 'serve-static';
 import helmet from 'helmet';
+import version from './version';
 import config from '../config';
 import { noRobots } from './no-robots';
 import log from './logging';
+import csp from '../lib/csp';
+import cspBlocking from '../lib/csp/blocking';
+import cspReportOnly from '../lib/csp/report-only';
 
+const app = express();
 const logger = log('server.main');
 const proxyUrl = config.get('proxyStaticResourcesFrom');
+const cspRulesBlocking = cspBlocking(config);
+const cspRulesReportOnly = cspReportOnly(config);
 
-const app = express();
-
-// TO DO: CSP, CORS, other security-related tasks (#4312)
+logger.info('version', { version: version });
 
 app.use(
   helmet.frameguard({
     action: 'deny',
   }),
   helmet.xssFilter(),
   helmet.noSniff(),
-  noRobots as express.RequestHandler
+  noRobots as express.RequestHandler,
+  bodyParser.text({
+    type: 'text/plain',
+  }),
+  bodyParser.json({
+    type: ['json', '*/json', 'application/csp-report'],
+  })
 );
 
+if (config.get('csp.enabled')) {
+  // @ts-ignore
```

Comment:
Most of these `@ts-ignore`s can probably be addressed somehow, does `as express.RequestHandler` fail? Might want a TODO around addressing those if they're being a pain.